content/en/docs/community/release_checklist.md

```diff
@@ -387,31 +408,70 @@ we're not all robots.
 You should also double check the URLs and checksums are correct in the
 auto-generated release notes.
 
-Once finished, go into GitHub and edit the release notes for the tagged release
-with the notes written here.
+Once finished, go into GitHub to [helm/helm
+releases](https://github.com/helm/helm/releases) and edit the release notes for
+the tagged release with the notes written here.
 
 It is now worth getting other people to take a look at the release notes before
 the release is published. Send a request out to
 [#helm-dev](https://kubernetes.slack.com/messages/C51E88VDG) for review. It is
 always beneficial as it can be easy to miss something.
 
-When you are ready to go, hit `publish`.
```

Comment:
Moving "publish" to its own step, after the PGP signing of the downloads.

Reply:
I did some rearranging to move "publish" to its own step. Since it's a visible change to the process (adding a step), it would be great for @bacongobbler and @mattfarina to double-check that.